protected/humhub/modules/space/controllers/CreateController.php

```diff
@@ -134,7 +136,15 @@ public function actionModules($space_id)
     {
         $space = Space::find()->where(['id' => $space_id])->one();
 
-        if (count($space->getAvailableModules()) == 0) {
+        //search through all modules to find module not set as default
+        $hasModuleNotSetAsDefault = Setting::find()
```

Comment:
Wouldn't it be easier to filter the `$availableModules` not longer in the `view` and instead directly in the `Controller`.

https://github.com/humhub/humhub/blob/master/protected/humhub/modules/space/views/create/modules.php#L24-L27

Otherwise I would rather filter the modules using the conditions from the view if possible. An own DB query is more performant, but more difficult to maintain.

Reply:
Done! Thank's for the corrections

Reply:
Controllers are also not a good place for such logic, why not something like `getInstallableModules($container)` in the module manager or something ?

Reply:
Done! Thank's